core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -77,11 +75,9 @@ public static BlockInStream createLocalBlockInStream(long blockId, long blockSiz
     try {
       BlockWorkerClient blockWorkerClient =
           closer.register(context.createBlockWorkerClient(workerNetAddress));
-      LockBlockResource lockBlockResource =
-          closer.register(blockWorkerClient.lockBlock(blockId, LockBlockOptions.defaults()));
       PacketInStream inStream = closer.register(PacketInStream
-          .createLocalPacketInStream(lockBlockResource.getResult().getBlockPath(), blockId,
-              blockSize));
+          .createLocalPacketInStream(context, blockWorkerClient.getDataServerAddress(), blockId,
+              blockWorkerClient.getSessionId(), blockSize));
```

Comment:
Yeah. That requires to have the heartbeat logic ready. My  PR is to make sure the short circuit read is similar as netty remote read. When you deprecate the session for block worker, you can remove htis